src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -252,6 +233,44 @@ public OperationalResponse checkout(OperationalRequest request) {
 		resp.setConfirm(true);
 		return resp;
 	}
+
+	public ResponseEntity<?> totalFavesAndCheckouts(int userId) {
+		OperationalResponse confirm = new OperationalResponse();
+		if(userRepository.getUserCount(userId) > 0) {
+			int totalCheckouts = userRepository.getCheckoutCount(userId);
+			List<User> userFavorites = userRepository.getUserFavorites(userId);
+			List<Favorites> favorites = curateFavorites(userFavorites);
+			if(favorites == null) {
+				confirm.setMessage("Could not retrieve movie title");
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).body(confirm);
+			}
+			FavesAndCheckOuts favesAndCheckOuts = new FavesAndCheckOuts();
+			favesAndCheckOuts.setCheckouts(totalCheckouts);
+			favesAndCheckOuts.setFavorites(favorites);
+			return ResponseEntity.status(HttpStatus.OK).body(favesAndCheckOuts);
+		}
+		confirm.setMessage( "User does not exist");
+		return ResponseEntity.status(HttpStatus.NOT_FOUND).body(confirm);
+	}
+
+	public List<Favorites> curateFavorites(List<User> userFavorites) {
+		System.out.println("::: " + userFavorites.size());
+		List<Favorites> favorites = new ArrayList<>();
+		for(User u : userFavorites) {
+			Favorites favorite = new Favorites();
+			SearchMoviesResponse searchAPIResp = APIClient.getAllMovies(new HashSet<>(Arrays.asList(u.getId().getProductId())));
+			if(searchAPIResp == null) {
+				return null;
+			}
+			favorite.setMovieId(u.getId().getProductId());
+			favorite.setMovieName(searchAPIResp.getResults().get(0).getTitle());
```

Comment:
In this case if the search team returns empty list `get(0)` will throw index out of bound exception. Although, if you do it the above suggested you might not need to hardcode `get(0)`